src/neuroconv/tools/nwb_helpers/_configure_backend.py

```diff
@@ -44,14 +46,29 @@ def configure_backend(
 
         neurodata_object = nwbfile.objects[object_id]
         is_dataset_linked = isinstance(neurodata_object.fields.get(dataset_name), TimeSeries)
+        if isinstance(neurodata_object.fields.get(dataset_name), AbstractDataChunkIterator):
+            data_chunk_iterator_class = None
+            data_chunk_iterator_kwargs = dict()
+        else:
+            data_chunk_iterator_class = DataChunkIterator
+            data_chunk_iterator_kwargs = dict(buffer_size=np.prod(dataset_configuration.buffer_shape))
```

Comment:
use math and not np.prod as that will overflow in most cases and screw up your buffer case. 

Are these changes related to repacking?

Reply:
> use math and not np.prod as that will overflow in most cases and screw up your buffer case.

Ok, will do.

> Are these changes related to repacking?

Yes, the DataChunkIterator solves this issue: https://github.com/hdmf-dev/hdmf/issues/1170

Reply:
Ah, thanks, OK. so hdmf does not support re-setting the chunking without copying so we do this through an iterator.

Mmm, you are solving your problem and this is great but in this PR as well as deteting compound data types I kind of feel that hdmf is kicking its complexity to us for something that it should provide. I guess that we are faster : P

Reply:
Speaking of which, I ran into another issue with hdmf-zarr when it comes to compound dtypes -- see here: https://github.com/hdmf-dev/hdmf-zarr/issues/272